### PR TITLE
[docs] docs: Record MoE overlap smoke results

### PR DESCRIPTION
**Pull request by yaoyu-33**
## Summary

Record measured MoE EP-overlap smoke results from short Qwen3 30B A3B pretrain experiments and update the related docs/skills with conservative caveats.

Current-main rerun on the same focused shape:

| Case | Knobs | Steady mean, iters 3-8 | Relative |
|---|---|---:|---:|
| A0 | alltoall, no EP overlap | 41.25s | 1.000x |
| A1 | EP overlap | 31.31s | 1.317x |
| A2 | EP overlap + delayed wgrad | 31.20s | 1.322x |

Key caveat: this remains a short smoke on the available PyTorch 25.11 / TE / Triton stack, and the measured runs keep `model.moe_permute_fusion=false` because the fused MoE permutation path failed in prior bring-up. The docs phrase the evidence as workload-specific and recommend rerunning on a branch-matched production container before generalizing.

## Files changed

- `docs/training/communication-overlap.md`
- `skills/perf-moe-comm-overlap/SKILL.md`
- `skills/perf-expert-parallel-overlap/SKILL.md`

## Validation

- `UV_CACHE_DIR=.local/uv-cache uv run pre-commit run --all-files`
- `git diff --check`
- No local unit tests run, per handoff.

## Notes

The detailed experiment report remains local under `.local/` and is not committed because it contains environment-specific cluster paths.

### CODE REVIEW

**Comment by yaoyu-33** [comment]
Current-main MoE EP-overlap rerun completed.

Context:
- Branch base: rebased onto latest `origin/main`, not stacked on #3867 or #3868
- Final SHA: `5b15b6b8691184d45204b478f0a6b4b3ddc9bed9`
- Workload: Qwen3 30B A3B pretrain, H100 BF16, config variant `v1`
- Scale: 2 nodes, 8 GPUs/node, 16 GPUs total, `EP=16`
- Fixed across rows: model/layout/batch/container/dispatcher/routing/CUDA graph setting
- Common overrides: `--moe_flex_dispatcher_backend None`, `--moe_a2a_overlap false`, `--cuda_graph_impl none`, `--tokenizer_type NullTokenizer`, `model.moe_shared_expert_overlap=false`, `model.moe_permute_fusion=false`
- Local unit tests: not run, per handoff
- Pre-commit: `UV_CACHE_DIR=.local/uv-cache uv run pre-commit run --all-files` passed

Result table, steady window = iterations 3-8:

| Case | Job | Effective overlap knobs | Status | Mean step | Median step | Mean TFLOP/s/GPU | Final loss | Rank-0 peak max allocated | Train log |
|---|---:|---|---|---:|---:|---:|---:|---:|---|
| `A0_baseline` | `11874554` | `overlap=false`, `delay_wgrad=false` | completed | 41.25s | 41.25s | 146.2 | 10.34087 | 46.068 GB | `/lustre/fsw/IOR-test/yuya/moe_ep_overlap_20260518/logs/A0_baseline_11874554_train.log` |
| `A1_overlap` | `11874557` | `overlap=true`, `delay_wgrad=false` | completed | 31.31s | 31.18s | 192.7 | 10.34093 | 46.750 GB | `/lustre/fsw/IOR-test/yuya/moe_ep_overlap_20260518/logs/A1_overlap_11874557_train.log` |
| `A2_overlap_delay` | `11874558` | `overlap=true`, `delay_wgrad=true` | completed | 31.20s | 31.18s | 193.3 | 10.34105 | 46.750 GB | `/lustre/fsw/IOR-test/yuya/moe_ep_overlap_20260518/logs/A2_overlap_delay_11874558_train.log` |

Overlap evidence:
- Baseline config: `/lustre/fsw/IOR-test/yuya/moe_ep_overlap_20260518/configs/A0_baseline_11874554.yaml`
  - `overlap_moe_expert_parallel_comm: false`
  - `delay_wgrad_compute: false`
- Overlap-only config: `/lustre/fsw/IOR-test/yuya/moe_ep_overlap_20260518/configs/A1_overlap_11874557.yaml`
  - `overlap_moe_expert_parallel_comm: true`
  - `delay_wgrad_compute: false`
  - `moe_shared_expert_overlap: false`
- Delayed-wgrad config: `/lustre/fsw/IOR-test/yuya/moe_ep_overlap_20260518/configs/A2_overlap_delay_11874558.yaml`
  - `overlap_moe_expert_parallel_comm: true`
  - `delay_wgrad_compute: true`
  - `moe_shared_expert_overlap: false`
- All rows kept `moe_token_dispatcher_type: alltoall`, `cuda_graph_impl: none`, `cuda_graph_scope: []`, `TP=1`, `PP=1`, `CP=1`, `EP=16`, `GBS=1024`, `MBS=1`.

Caveats:
- Short mock-data smoke only; not a release-grade throughput claim.
- `model.moe_permute_fusion=false` remains a runtime-stack workaround for this container.
- The all-to-all dispatcher was fixed intentionally because the prior dispatcher smoke found missing DeepEP/HybridEP packages in this container.
- Delayed wgrad was effectively neutral versus overlap-only on this shape: 31.20s vs 31.31s.

**Review by cuichenx — APPROVED** [review]
